include/zephyr/platform/platform.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_PLATFORM_PLATFORM_H_
+#define ZEPHYR_INCLUDE_PLATFORM_PLATFORM_H_
+
+
+void platform_early_initialize(void);
+void platform_late_initialize(void);
```

Comment:
since these are called internally, they should not be exposed in a public header. we should likely think about internal headers... now everything is under `include/zephyr`, sometimes you can see `header_internal.h`, but it's not a strict rule.

Reply:
> since these are called internally, they should not be exposed in a public header. we should likely think about internal headers... now everything is under `include/zephyr`, sometimes you can see `header_internal.h`, but it's not a strict rule.

internal to what? The call is internal, but the implementation can be external to zephyr, so we need to let implementers know what API to implement. So we need to somehow expose this as an interface (public) that can be implemented and not called somehow. We will have the same issue with other hooks, so we need to figure out how to position it and expose it.

Reply:
Only Kernel is meant to call these hooks, so placing them in `zephyr/platform.h` will make people feel they can include that file and call the functions somewhere else.

Reply:
yes, I understand, let me think about it a bit more.

Reply:
some googling reveals others with similar approach :)

https://github.com/apache/nuttx/blob/master/include/nuttx/board.h

https://github.com/apache/nuttx/blob/a73f5f86dcb05ec98d66f70966f02624f2e2c9f3/include/nuttx/board.h#L103:

> * Public Function Prototypes
>  *
> * These are all standard board interfaces that are exported from board-
>  * specific logic to OS internal logic.  These should never be accessed
>  * directly from application code but may be freely used within the OS
>  *

We could have something similar,

Reply:
Zephyr looking at NuttX :open_mouth: (just joking, I think NuttX has many well-developed concepts)